connection.lua

```diff
@@ -116,7 +116,7 @@ function Connection:_changeState(to, data)
 end
 
 function Connection:_error(err)
-  self._log(logging.ERROR, tostring(err))
+  self._log(logging.WARNING, tostring(err))
```

Comment:
Feels like it would be better to create a `_warning` method and switch some things around to use it.

Something like this should still be an error condition rather than warning - https://github.com/virgo-agent-toolkit/virgo-base-agent/blob/0513c8aa3cc0b774b389837cd1678a8ffa66eee4/connection.lua#L142-L144

Reply:
Ha, I didn't even notice the method is called `_error`. Yeah, I'll see if I can change along the lines of your suggestion.